### PR TITLE
Add initial documentation for the compass

### PR DESCRIPTION
**Pull request by deshipu**

### CODE REVIEW

**Comment by deshipu** [comment]
This is an initial version, to get started.

It misses the descriptions of the format in which get_?() and heading() functions return what they return.

File: docs/compass.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+
+.. py:function:: clear_calibrarion()
+
+    Undoes the calibration, making the compass uncalibrated again.
+
+
+.. py:function:: get_x()
+
+    Gives the reading on the ``x`` axis of the compass.
```
Comment (ntoll) [inline docs/compass.rst]: ... as an integer.

File: docs/compass.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+.. py:function:: get_y()
+
+    Gives the reading on the ``y`` axis of the compass.
```
Comment (ntoll) [inline docs/compass.rst]: ... as an integer.

File: docs/compass.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+.. py:function:: get_z()
+
+    Gives the reading on the ``z`` axis of the compass (vertical).
```
Comment (ntoll) [inline docs/compass.rst]: ... as an integer.

File: docs/compass.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
+.. py:function:: heading()
+
+    Gives the compass heading, calculated from the above readings.
```
Comment (ntoll) [inline docs/compass.rst]: Gives the compass heading as an integer representation of degrees (0-359) with north as zero. Calculated from the x, y and z readings.

**Comment by ntoll** [comment]
Hope my comments add a bit more context for revisions.

File: docs/compass.rst (hunk 1 above)
Comment (deshipu) [inline docs/compass.rst]: What is the range? What does it represent?

File: docs/compass.rst (hunk 4 above)
Comment (deshipu) [inline docs/compass.rst]: Great, I will add that.

Comment (deshipu) [inline docs/compass.rst]: Hmm, clockwise or counter-clockwise?

File: docs/compass.rst (hunk 1 above)
Comment (ntoll) [inline docs/compass.rst]: I have no idea of the range but it returns +/- values and represents the magnetic force measured along the referenced axis.

File: docs/compass.rst (hunk 4 above)
Comment (ntoll) [inline docs/compass.rst]: Clockwise (I just checked). 

Strangely enough, the range is 0-360. :-/ This is probably a bug in the upstream device abstraction layer.

**Pull request merged**